### PR TITLE
Include millisatoshis in queryroutes

### PR DESCRIPTION
**Pull request by t4sk**
`queryroutes` return values rounded off to Satoshi. This would cause
payments to have insufficient fee when attempting to use the output of
`queryroutes` as an input to [sendtoroute](https://github.com/lightningnetwork/lnd/pull/747).
 
This PR modifies the response of `queryroutes` to include fees and amounts in millisatoshis

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -1152,8 +1152,10 @@ message Hop {
     uint64 chan_id = 1 [json_name = "chan_id"];
     int64 chan_capacity = 2 [json_name = "chan_capacity"];
     int64 amt_to_forward = 3 [json_name = "amt_to_forward"];
-    int64 fee = 4 [json_name = "fee"];
-    uint32 expiry = 5 [json_name = "expiry"];
+    int64 amt_to_forward_msat = 4 [json_name = "amt_to_forward_msat"];
+    int64 fee = 5 [json_name = "fee"];
+    int64 fee_msat = 6 [json_name = "fee_msat"];
+    uint32 expiry = 7 [json_name = "expiry"];
```
Comment (Roasbeef) [inline lnrpc/rpc.proto]: In order to not break the set of existing protos, all these fields need to be added at the _end_.

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -1180,19 +1182,29 @@ message Route {
...
     satoshis, otherwise the route will fail at an intermediate node due to an
     insufficient amount of fees.
     */
-    int64 total_amt = 3 [json_name = "total_amt"];
+    int64 total_amt = 4 [json_name = "total_amt"];
+
+    /**
+    The total amount in millisatoshis.
+    */
+    int64 total_amt_msat = 5 [json_name = "total_amt_msat"];
```
Comment (Roasbeef) [inline lnrpc/rpc.proto]: Similar comment here w.r.t breaking the existing set of protos.

File: lnrpc/rpc.proto [hunk 3]
```diff
@@ -1154,6 +1154,8 @@ message Hop {
     int64 amt_to_forward = 3 [json_name = "amt_to_forward"];
     int64 fee = 4 [json_name = "fee"];
     uint32 expiry = 5 [json_name = "expiry"];
+    int64 amt_to_forward_msat = 6 [json_name = "amt_to_forward_msat"];
```
Comment (halseth) [inline lnrpc/rpc.proto]: You can add `[deprecated=true]` to the satoshi fields, per https://developers.google.com/protocol-buffers/docs/proto3

File: lnd_test.go [hunk 4]
```diff
@@ -1384,6 +1384,195 @@ func assertPendingHtlcStageAndMaturity(t *harnessTest,
...
+				i, route.TotalFeesMsat, route.TotalFees)
+		}
+		if route.TotalFeesMsat != int64(expectedTotalFeesMSat) {
+			t.Fatalf("route %v: total fees in msat expected %v got %v",
+				i, expectedTotalFeesMSat, route.TotalFeesMsat)
+		}
+
+		// check total amount
+		if route.TotalAmt != route.TotalAmtMsat/mSat {
+			t.Fatalf("route %v: total amt %v (msat) does not round down to %v (sat)",
```
Comment (halseth) [inline lnd_test.go]: lines look a bit longer than 80 chars.

File: lnd_test.go [hunk 5]
```diff
@@ -1384,6 +1384,195 @@ func assertPendingHtlcStageAndMaturity(t *harnessTest,
...
+				"down to %v (sat)",
+				i, len(route.Hops)-1, hop.AmtToForwardMsat, hop.AmtToForward)
+		}
+		if hop.AmtToForwardMsat != paymentAmt*mSat {
+			t.Fatalf("route %v hop %v: amt to forward in msat expected %v got %v",
+				i, len(route.Hops)-1, paymentAmt*mSat, hop.AmtToForwardMsat)
+		}
+	}
+
+	// clean up
```
Comment (halseth) [inline lnd_test.go]: nit: Comments should be full sentences.

File: lnd_test.go [hunk 6]
```diff
@@ -7938,6 +8127,10 @@ type testCase struct {
 }
 
 var testsCases = []*testCase{
+	{
+		name: "query routes",
```
Comment (halseth) [inline lnd_test.go]: add to the end of the test case list or close to the test it is modelled of.

File: lnd_test.go (hunk 4 above)
Comment (t4sk) [reply lnd_test.go]: Do you count a tab as 4 spaces?

**Comment by Roasbeef** [comment]
Can you rebase this to resolve the proto conflict? Really looking forward to finally getting `sendroute` merged!

**Comment by t4sk** [comment]
Rebased

**Pull request merged**